corehq/apps/userreports/tests/test_app_manager_integration.py

```diff
@@ -1,17 +1,44 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
-import json
 import os
 from datetime import datetime
+
+from django.test import TestCase
 from mock import patch, Mock
-from django.test import SimpleTestCase
-from corehq.apps.app_manager.models import Application
+from corehq.apps.app_manager.models import Application, Module
+from corehq.apps.app_manager.tests.app_factory import AppFactory
 from corehq.apps.userreports.app_manager.helpers import get_case_data_sources, get_form_data_sources
 from corehq.apps.userreports.reports.builder import DEFAULT_CASE_PROPERTY_DATATYPES
 from dimagi.utils.parsing import json_format_datetime
 
 
-class AppManagerDataSourceConfigTest(SimpleTestCase):
+class AppManagerDataSourceConfigTest(TestCase):
+    domain = 'userreports_test'
+    case_type = 'app_data_case'
+    case_properties = ['first_name', 'last_name', 'children', 'dob']
+
+    @classmethod
+    def setUpClass(cls):
+        super(AppManagerDataSourceConfigTest, cls).setUpClass()
+        cls.app = Application.new_app(cls.domain, 'Application Data Source App')
```

Comment:
Lovely, these kind of tests are so much easier to work with than the json-based ones. I usually see the whole app created with `AppFactory`, curious is there's a reason to prefer making it with `Application` here?

Reply:
Also curious about this. `AppFactory` does pretty much the same thing as what you have done here (make a new app, add module, etc). :+1: to the test conversion though

Reply:
It didn't look like AppFactory had any way to easily specify a source file for the form (which is needed for these tests). Now that I say that I could probably add that in though.

Reply:
```
m0, f0 = factory.new_basic_module('Untitled Module', cls.case_type)
f0.source = cls.get_xml('simple')
```

Reply:
gracias.

Reply:
https://github.com/dimagi/commcare-hq/pull/19987/commits/52671f706221ad77f3e1555254b4e79b667e3237